arelle/BetaFeatures.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import annotations
+
+BETA_OBJECT_MODEL_FEATURE = "betaObjectModel"
+# Add camelCaseOptionName
+BETA_FEATURES_AND_DESCRIPTIONS: dict[str, str] = {
+    BETA_OBJECT_MODEL_FEATURE: "Replace lxml based object model with a pure Python class hierarchy.",
+}
+
+BETA_FEATURES = BETA_FEATURES_AND_DESCRIPTIONS.keys()
+
+
+_NEW_OBJECT_MODEL_STATUS_ACCESSED = False
+_USE_NEW_OBJECT_MODEL = False
+
+
+def enableNewObjectModel() -> None:
+    global _USE_NEW_OBJECT_MODEL
+    if _USE_NEW_OBJECT_MODEL:
+        return
+    if _NEW_OBJECT_MODEL_STATUS_ACCESSED:
+        raise RuntimeError("Can't change object model transition setting after classes have already been defined.")
+    _USE_NEW_OBJECT_MODEL = True
+
+
+def newObjectModelEnabled() -> bool:
+    global _NEW_OBJECT_MODEL_STATUS_ACCESSED
+    _NEW_OBJECT_MODEL_STATUS_ACCESSED = True
```

Comment:
I would concur if this situation involved a handled use case that provided a warning or offered the user with alternative options for continuing. If we ever attempt to activate the new models after another part of the app believes we're using the old models, it's crucial to trigger a loud and immediate error with a clear and distinct exception. This approach achieves that without requiring all parts of Arelle that use the setting to take on that responsibility.

Reply:
I aim to assert two things here:

1. Models are not created with a mix of base classes.
2. If the beta feature is enabled, the models are created with the new base classes.

The first issue appears to be addressed by the inheritance assertions in the unit tests. Since `Cntlr.__init__` is not the only way models can be initialized, this seems like a reasonable approach to me.

The second issue is quite fragile due to the conditional import in the models init module. If the models are imported before the beta feature is enabled, the feature will be ignored. This could easily happen inadvertently if imports are added or reordered.

Reply:
That's fair. It would require explicit handling of setting false for the GUI and library paths (and testing of the frozen builds) where this feature can't be enabled anyway. I'm inclined to leave this as is unless you feel strongly.